File: ABC/142/a.py
n = int(input())

if n == 1:
    print('{0:.10f}'.format(1))
else:
    a = n%2
    if a == 1:
        amari = ((n-1)/2 +1)/n
        print('{0:.10f}'.format(amari))
    else:
        print('{0:.10f}'.format(0.5))

File: ABC/142/a.cpp
#include <iostream>

using namespace std;

int main()
{
  int n;
  cin >> n;
  int a;
  a = n%2;
  std::cout << a << '\n';
  if(a==1)
  {
    double amari = ((n-1)/2+1)/n;
    std::cout << amari << '\n';
  }
  else
  {
    double amari = 0.5;
    std::cout << amari << '\n';
  }

}
